netplan_cli/cli/utils.py

```diff
@@ -196,10 +196,43 @@ def get_interface_driver_name(interface, only_down=False):  # pragma: nocover (c
     return driver_name
 
 
-def get_interface_macaddress(interface):
-    # return an empty list (and string) if no LL data can be found
-    link = netifaces.ifaddresses(interface).get(netifaces.AF_LINK, [{}])[0]
-    return link.get('addr', '')
+def _get_permanent_macaddress(interface: str) -> str:
+    mac = ""
+    try:
+        out = subprocess.check_output(['ethtool', '-P', interface]).decode('utf-8')
```

Comment:
note (non-blocking): We need a new runtime dependency on `ethtool` to make this work. (It's already in Ubuntu "main")

Could you please temporarily add it to the CI test, so we can see if this changes behaviour in any way? For now it would just be falling back to non-permanent MAC in case `ethtool` is not installed. – Which is fine, IMO, but we want to test the real thing in the integration tests.

Reply:
ethtool is seeded on Ubuntu actually. We might need to install it for the Debian test in our CI though.

Reply:
https://salsa.debian.org/debian/netplan.io/-/commit/d6405858d585b833eb5d1c58cf43eff8568a16ed